src/aui/framemanager.cpp

```diff
@@ -1268,16 +1291,18 @@ wxString wxAuiManager::SavePerspective()
 bool wxAuiManager::LoadPerspective(const wxString& layout, bool update)
 {
     wxString input = layout;
-    wxString part;
+    wxString layoutVersion;
 
     // check layout string version
     //    'layout1' = wxAUI 0.9.0 - wxAUI 0.9.2
     //    'layout2' = wxAUI 0.9.2 (wxWidgets 2.8)
-    part = input.BeforeFirst(wxT('|'));
+    //    'layout3' = wxWidgets 3.3.1
+    layoutVersion = input.BeforeFirst(wxT('|'));
     input = input.AfterFirst(wxT('|'));
-    part.Trim(true);
-    part.Trim(false);
-    if (part != wxT("layout2"))
+    layoutVersion.Trim(true);
+    layoutVersion.Trim(false);
+    if (layoutVersion != wxT("layout3") &&
```

Comment:
This is a typo, right? Should be

```suggestion
    if (layoutVersion != wxT("layout2") &&
```

Reply:
You are correct.  See https://github.com/wxWidgets/wxWidgets/pull/25483/commits/aab24ad28074e180d6e5a5ffcafc90aa57989179